app/src/main/java/org/sopt/dosopttemplate/MainActivity.kt

```diff
@@ -1,11 +1,92 @@
 package org.sopt.dosopttemplate
 
-import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import android.widget.Toast
+import androidx.activity.ComponentActivity
+import androidx.activity.compose.setContent
+import androidx.activity.viewModels
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.lifecycle.flowWithLifecycle
+import androidx.lifecycle.lifecycleScope
+import androidx.navigation.NavHostController
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import androidx.navigation.compose.rememberNavController
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import org.sopt.dosopttemplate.feature.home.HomeScreen
+import org.sopt.dosopttemplate.feature.login.LoginScreen
+import org.sopt.dosopttemplate.feature.signup.SignUpScreen
+import org.sopt.dosopttemplate.ui.theme.DoEuijinKwakTheme
+
+class MainActivity : ComponentActivity() {
+
+    enum class Screen {
+        LOGIN,
+        SIGNUP,
+        Home,
+    }
+
+    private val viewModel by viewModels<MainViewModel>()
 
-class MainActivity : AppCompatActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_main)
+        setContent {
+            DoEuijinKwakTheme {
+                val navController = rememberNavController()
```

Comment:
```suggestion
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContent {
            DoEuijinKwakTheme {
                val navController = rememberNavController()
                val viewModel: MainViewModel = viewModel()
```

컴포즈에서는 delegate 방식이 아닌 함수 내에서 변수 선언방식으로 뷰모델을 불러올 수도 있습니다.

Reply:
오호 처음알았네요 그러면 저 viewmodel은 어떤 생명주기를 가지고있나요??